low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -285,3 +309,142 @@ def get_fans_info(self):
+        try:
+            unit = self._bus.get_object(
+                SYSTEMD_BUS, self._manager.LoadUnit(service_name))
+            properties_iface = Interface(unit, dbus_interface=PROPERTIES_IFACE)
+        except DBusException:
+            # TODO add logs
+            return None
+        path_array = properties_iface.Get(SERVICE_IFACE, 'ExecStart')
+        try:
+            command_line_path = str(path_array[0][0])
+        except IndexError:
+            # TODO add logs
+            command_line_path = "NA"
+
+        is_installed = True if command_line_path != "NA" or 'invalid' in properties_iface.Get(
+            UNIT_IFACE, 'UnitFileState') else False
+        uid = str(properties_iface.Get(UNIT_IFACE, 'Id'))
+        if not is_installed:
+            health_status = "NA"
+            health_description = f"{uid} is not installed"
+            recommendation = "NA"
+            specifics = [
+                {
+                    "service_name": uid,
+                    "description": "NA",
+                    "installed": str(is_installed).lower(),
+                    "pid": "NA",
+                    "state": "NA",
+                    "substate": "NA",
+                    "status": "NA",
+                    "license": "NA",
+                    "version": "NA",
+                    "command_line_path": "NA"
+                }
+            ]
+        else:
+            service_description = str(
+                properties_iface.Get(UNIT_IFACE, 'Description'))
+            state = str(properties_iface.Get(UNIT_IFACE, 'ActiveState'))
+            substate = str(properties_iface.Get(UNIT_IFACE, 'SubState'))
+            service_status = 'enabled' if 'disabled' not in properties_iface.Get(
+                UNIT_IFACE, 'UnitFileState') else 'disabled'
+            pid = "NA" if state == "inactive" else str(
+                properties_iface.Get(SERVICE_IFACE, 'ExecMainPID'))
+            version = self.get_service_info_from_rpm(uid, "VERSION")
+            service_license = self.get_service_info_from_rpm(uid, "LICENSE")
+            specifics = [
+                {
+                    "service_name": uid,
+                    "description": service_description,
+                    "installed": str(is_installed).lower(),
+                    "pid": pid,
+                    "state": state,
+                    "substate": substate,
+                    "status": service_status,
+                    "license": service_license,
+                    "version": version,
+                    "command_line_path": command_line_path
+                }
+            ]
+            if service_status == 'enabled' and state == 'active' \
+                    and substate == 'running':
+                health_status = 'OK'
+                health_description = f"{uid} is in good health"
+                recommendation = "NA"
+            else:
+                health_status = "substate"
```

Comment:
why feeding substate here

Reply:
Sorry, this should we substate variable, removed the quotes.

Reply:
Ok. still, why not using 'state', & using sub-state ?

Reply:
As per [dbus documentation](https://www.freedesktop.org/wiki/Software/systemd/dbus/), 
_SubState encodes states of the same state machine that ActiveState (from where state is coming) covers but knows more fine-grained states that are unit-type-specific._ 
But yes, not all ActiveState have corresponding SubState. 
I will change this to state.

Reply:
Changed to state